HarmonyOS/Day3 Assingment/KanishkaHalder/LoginSignUpApp/entry/src/main/java/com/applib/loginsignupapp/slice/SignUpSlice.java

```diff
@@ -0,0 +1,111 @@
+import ohos.aafwk.ability.DataAbilityHelper;
+import ohos.aafwk.ability.DataAbilityRemoteException;
+import ohos.aafwk.ability.OnClickListener;
+import ohos.aafwk.content.Intent;
+import ohos.agp.components.Button;
+import ohos.agp.components.Component;
+import ohos.agp.components.RadioButton;
+import ohos.agp.components.RadioContainer;
+import ohos.agp.window.dialog.ToastDialog;
+import ohos.data.DatabaseHelper;
+import ohos.data.orm.OrmContext;
+import ohos.data.rdb.ValuesBucket;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+import ohos.utils.net.Uri;
+
+public class SignUpSlice extends AbilitySlice {
+
+    private static final Uri URI =  Uri.parse("dataability:///com.applib.loginsignupapp.DataAbility/test");
+    private static final HiLogLabel LABEL_LOG = new HiLogLabel(HiLog.DEBUG, 0xD001100, "SIGNUP_LOG");
+    private DataAbilityHelper dbHelper;
+
+    private TextFieldValidated firstName;
+    private TextFieldValidated lastName;
+    private TextFieldValidated emailId;
+    private TextFieldValidated phoneNumber;
+    private TextFieldValidated password;
+    private Button submitBtn;
+    private RadioContainer radioContainer;
+    @Override
+    protected void onStart(Intent intent) {
+        super.onStart(intent);
+        super.setUIContent(ResourceTable.Layout_ability_signup);
+//        "permissions": [
+//          "com.applib.loginsignupapp.DataAbilityShellProvider.PROVIDER"
+//        ],
+        dbHelper = DataAbilityHelper.creator(this,URI);
+
+
+        firstName = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_firstName);
+        lastName = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_lastName);
+        emailId = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_email);
+        phoneNumber = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_phone);
+        password = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_password);
+        submitBtn = (Button) findComponentById(ResourceTable.Id_button_signup_signup_slice);
+        radioContainer = (RadioContainer) findComponentById(ResourceTable.Id_radio_container);
+        submitBtn.setClickedListener(new Component.ClickedListener() {
+            @Override
+            public void onClick(Component component) {
+
+                boolean isAllValidated = true;
+                ValuesBucket value = new ValuesBucket();
+                if(firstName.getText()!=null) {
+                    value.putString("firstName", firstName.getText());
+                    isAllValidated = firstName.isValidated() && isAllValidated;
+                }
+                if(lastName.getText()!=null) {
+                    value.putString("lastName", lastName.getText());
+                    isAllValidated = lastName.isValidated() && isAllValidated;
+                }
+                if(emailId.getText()!=null) {
+                    value.putString("email", emailId.getText());
```

Comment:
Avoid using hardcoded strings. Use constants instead. (Applies to all the files)